eclipse-collections-code-generator/src/main/resources/impl/map/immutable/immutablePrimitiveObjectHashMap.stg

```diff
@@ -866,6 +869,20 @@ final class Immutable<name>ObjectHashMap\<V> extends AbstractImmutable<name>Obje
     }
 
     @Override
+    public ImmutableObject<name>Map\<V> flipUniqueValues()
+    {
+        MutableObject<name>Map\<V> result = Object<name>Maps.mutable.empty();
+        this.forEachKeyValue((key, value) -> {
+            if (result.containsKey(value))
+            {
+                throw new IllegalStateException("Duplicate value: " + value + " found at key: " + result.get(value) + " and key: " + key);
+            }
+            result.put(value, key);
```

Comment:
We don't have any API that would let us calculate the hashcode just once?

Reply:
Not that I am aware of